src/models/action.rs

```diff
@@ -0,0 +1,95 @@
+use chrono::NaiveDateTime;
+use diesel::prelude::*;
+use diesel::{
+    deserialize::{self, FromSql},
+    pg::Pg,
+    serialize::{self, Output, ToSql},
+    sql_types::Integer,
+};
+use std::io::Write;
+
+use crate::models::{ApiToken, User, Version};
+use crate::schema::*;
+
+#[derive(Debug, Clone, Copy, FromSqlRow, AsExpression)]
+#[repr(u32)]
```

Comment:
What do you think about making this `#[repr(i32)]` instead? Then we can drop the `Into` impl and write `self as i32` instead.

Reply:
Done